### PR TITLE
Feature Parity for Open Embedded

### PR DESCRIPTION
**Pull request by allenh1**
This is the Open Embedded side of #76's changes.

### CODE REVIEW

**Comment by allenh1** [comment]
Commit 2a24583 resolves #62.

File: superflore/generators/bitbake/ros_meta.py [hunk 1]
```diff
@@ -12,67 +12,63 @@
...
 
-
-# TODO(allenh1): remove inheritance like the RosOverlay class.
+import os
 import random
 import string
 import time
 
-from superflore import RepoInstance as repo_instance
+from superflore.generators.ebuild.overlay_instance import get_random_tmp_dir
```
Comment (tfoote) [inline superflore/generators/bitbake/ros_meta.py]: This looks like a chance for better sharing by moving to utils?

File: superflore/generators/bitbake/ros_meta.py [hunk 2]
```diff
@@ -12,67 +12,63 @@
...
+class RosMeta(object):
+    def __init__(self, repo_dir=None):
         # clone repo into a random tmp directory.
-        repo_instance.__init__(self, 'allenh1',
-                               'meta-ros', get_random_tmp_dir())
+        do_clone = True
+        if repo_dir:
+            do_clone = not os.path.exists(os.path.realpath(repo_dir))
+        self.repo = RepoInstance(
+            'allenh1', 'meta-ros', repo_dir or get_random_tmp_dir(),
```
Comment (tfoote) [inline superflore/generators/bitbake/ros_meta.py]: hard coded value here, at least make it an argument with a default

File: superflore/generators/bitbake/ros_meta.py (hunk 1 above)
Comment (allenh1) [reply superflore/generators/bitbake/ros_meta.py]: Yes, this was a consideration... I think I'll change it to have a function that returns a group of random characters of some length. That way I could just call it and inline the formatted directories.

Comment (tfoote) [reply superflore/generators/bitbake/ros_meta.py]: Sorry looking under the hood you should be using tempfile.mkdtemp And while you're doing that use a context manager to clean it afterwards. like this: https://gist.github.com/tfoote/f68ad3f6a6a04a41264f65a04ef86fac

**Comment by allenh1** [comment]
@tfoote I'm a little confused as to how I implement the TempfileManager. I understand I'm supposed to use a `with` statement, but I'm not sure where the `with` should go... Any insight here?

**Comment by Evidlo** [comment]
@allenh1 See [this](https://stackoverflow.com/questions/1984325/explaining-pythons-enter-and-exit)

**Comment by tfoote** [comment]
> @tfoote I'm a little confused as to how I implement the TempfileManager. I understand I'm supposed to use a with statement, but I'm not sure where the with should go... Any insight here?

Instead of doing 
```
mkdir foo
operate inside foo
rmdir foo
```
change it to:
```
with TempfileManager(foo) as tempdir:
    operate inside foo
```
When the with statement is finished the __exit__ method will be called and things will be cleaned up. Likely you'll want this in your top level logic (near the command line options) and the working directory will be passed into the build specific logic which won't know if it's temporary or passed via an argument to avoid cleanup. And they can use a implementation specific subdirectory of the tempdir.

**Comment by allenh1** [comment]
@tfoote nice tip! Took me a few to understand it, but it's super neat.

**Pull request merged**